### PR TITLE
Add com.moddb.TotalChaos

### PR DESCRIPTION
**Pull request by AsciiWolf**
Flatpak for the [Total Chaos](https://www.moddb.com/mods/total-chaos) game.

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build com.moddb.TotalChaos

File: com.moddb.TotalChaos.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com.moddb.TotalChaos.desktop</id>
```
Comment (TingPing) [inline com.moddb.TotalChaos.appdata.xml]:
```suggestion
  <id>com.moddb.TotalChaos</id>
  <launchable type="desktop-id">com.moddb.TotalChaos.desktop</launchable>
```

File: com.moddb.TotalChaos.desktop [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Name=Total Chaos
+Comment=Survival horror set on a remote island known as Fort Oasis
+GenericName=First Person Shooter Game
```
Comment (TingPing) [inline com.moddb.TotalChaos.desktop]: I don't think this is a useful generic name, probably just remove it.

File: com.moddb.TotalChaos.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    sha256: c90912b77f1b545d5a861a71098507328a01e81a697c3a69acf5777579f01dd1
+  - type: file
+    url: https://github.com/coelckers/gzdoom/raw/g3.6.0/soundfont/gzdoom.sf2
+    sha256: fca3e514b635a21789d4224e84865d2954a2a914d46b64aa8219ddb565c44869
+  - type: shell
+    commands:
+    - install -Dm 644 gzdoom.sf2 /app/share/sounds/sf2/gzdoom.sf2
+  - type: script
+    commands:
+    - gzdoom -file totalchaos.pk3 +fluid_patchset /app/share/sounds/sf2/gzdoom.sf2 $@
```
Comment (TingPing) [inline com.moddb.TotalChaos.yaml]:
```suggestion
      - gzdoom -file totalchaos.pk3 +fluid_patchset /app/share/sounds/sf2/gzdoom.sf2 $@
```

File: com.moddb.TotalChaos.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  sources:
+  - type: archive
+    url: https://github.com/coelckers/gzdoom/archive/g3.6.0.tar.gz
+    sha256: c90912b77f1b545d5a861a71098507328a01e81a697c3a69acf5777579f01dd1
+  - type: file
+    url: https://github.com/coelckers/gzdoom/raw/g3.6.0/soundfont/gzdoom.sf2
+    sha256: fca3e514b635a21789d4224e84865d2954a2a914d46b64aa8219ddb565c44869
+  - type: shell
+    commands:
+    - install -Dm 644 gzdoom.sf2 /app/share/sounds/sf2/gzdoom.sf2
```
Comment (TingPing) [inline com.moddb.TotalChaos.yaml]:
```suggestion
      - install -Dm 644 gzdoom.sf2 /app/share/sounds/sf2/gzdoom.sf2
```

File: com.moddb.TotalChaos.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    url: https://github.com/FluidSynth/fluidsynth/archive/v2.0.1.tar.gz
+    sha256: 0d80f0b2fb0d3f93722844e654d5b5f004bf31a07f752c0be939b7242ac33248
+
+- name: gzdoom
+  buildsystem: cmake-ninja
+  config-opts:
+  - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+  - -DSEND_ANON_STATS=OFF
+  sources:
+  - type: archive
```
Comment (TingPing) [inline com.moddb.TotalChaos.yaml]: Just indent all of this.. https://github.com/flathub/flathub/wiki/YAML-Style-Guide

**Comment by AsciiWolf** [comment]
@TingPing Should be fixed now.

**Comment by AsciiWolf** [comment]
bot, build com.moddb.TotalChaos

File: com.moddb.TotalChaos.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com.moddb.TotalChaos</id>
```
Comment (TingPing) [inline com.moddb.TotalChaos.appdata.xml]: Missing `launchable`

File: com.moddb.TotalChaos.yaml [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    sources:
+      - type: archive
+        url: https://github.com/FluidSynth/fluidsynth/archive/v2.0.1.tar.gz
+        sha256: 0d80f0b2fb0d3f93722844e654d5b5f004bf31a07f752c0be939b7242ac33248
+
+  - name: gzdoom
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      - -DSEND_ANON_STATS=OFF
```
Comment (TingPing) [inline com.moddb.TotalChaos.yaml]: If upstream wants this to be an official package perhaps they want this?

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.moddb.TotalChaos

**Comment by nedrichards** [comment]
@AsciiWolf to add the upstream developer they need to have a GitHub account - if you let us know what that is we'll happily add them.

**Comment by AsciiWolf** [comment]
@nedrichards I sent an e-mail to the upstream developer regarding this. Will let you know when I get an answer. Thanks!

**Pull request closed without merging**